src/sgp/TranslationTable.cc

```diff
@@ -909,15 +909,7131 @@ class CTranslationTableRussianGold : public CTranslationTable
+		m_table[40138] = 6881; // 鳊
+		m_table[40139] = 6882; // 鳋
+		m_table[40140] = 6883; // 鳌
+		m_table[40141] = 6884; // 鳍
+		m_table[40142] = 6885; // 鳎
+		m_table[40143] = 6886; // 鳏
+		m_table[40144] = 6887; // 鳐
+		m_table[40147] = 6888; // 鳓
+		m_table[40148] = 6889; // 鳔
+		m_table[40149] = 6890; // 鳕
+		m_table[40151] = 6891; // 鳗
+		m_table[40152] = 6892; // 鳘
+		m_table[40153] = 6893; // 鳙
+		m_table[40156] = 6894; // 鳜
+		m_table[40157] = 6895; // 鳝
+		m_table[40159] = 6896; // 鳟
+		m_table[40162] = 6897; // 鳢
+		m_table[38780] = 6898; // 靼
+		m_table[38789] = 6899; // 鞅
+		m_table[38801] = 6900; // 鞑
+		m_table[38802] = 6901; // 鞒
+		m_table[38804] = 6902; // 鞔
+		m_table[38831] = 6903; // 鞯
+		m_table[38827] = 6904; // 鞫
+		m_table[38819] = 6905; // 鞣
+		m_table[38834] = 6906; // 鞲
+		m_table[38836] = 6907; // 鞴
+		m_table[39601] = 6908; // 骱
+		m_table[39600] = 6909; // 骰
+		m_table[39607] = 6910; // 骷
+		m_table[40536] = 6911; // 鹘
+		m_table[39606] = 6912; // 骶
+		m_table[39610] = 6913; // 骺
+		m_table[39612] = 6914; // 骼
+		m_table[39617] = 6915; // 髁
+		m_table[39616] = 6916; // 髀
+		m_table[39621] = 6917; // 髅
+		m_table[39618] = 6918; // 髂
+		m_table[39627] = 6919; // 髋
+		m_table[39628] = 6920; // 髌
+		m_table[39633] = 6921; // 髑
+		m_table[39749] = 6922; // 魅
+		m_table[39747] = 6923; // 魃
+		m_table[39751] = 6924; // 魇
+		m_table[39753] = 6925; // 魉
+		m_table[39752] = 6926; // 魈
+		m_table[39757] = 6927; // 魍
+		m_table[39761] = 6928; // 魑
+		m_table[39144] = 6929; // 飨
+		m_table[39181] = 6930; // 餍
+		m_table[39214] = 6931; // 餮
+		m_table[39253] = 6932; // 饕
+		m_table[39252] = 6933; // 饔
+		m_table[39647] = 6934; // 髟
+		m_table[39649] = 6935; // 髡
+		m_table[39654] = 6936; // 髦
+		m_table[39663] = 6937; // 髯
+		m_table[39659] = 6938; // 髫
+		m_table[39675] = 6939; // 髻
+		m_table[39661] = 6940; // 髭
+		m_table[39673] = 6941; // 髹
+		m_table[39688] = 6942; // 鬈
+		m_table[39695] = 6943; // 鬏
+		m_table[39699] = 6944; // 鬓
+		m_table[39711] = 6945; // 鬟
+		m_table[39715] = 6946; // 鬣
+		m_table[40637] = 6947; // 麽
+		m_table[40638] = 6948; // 麾
+		m_table[32315] = 6949; // 縻
+		m_table[40578] = 6950; // 麂
+		m_table[40583] = 6951; // 麇
+		m_table[40584] = 6952; // 麈
+		m_table[40587] = 6953; // 麋
+		m_table[40594] = 6954; // 麒
+		m_table[37846] = 6955; // 鏖
+		m_table[40605] = 6956; // 麝
+		m_table[40607] = 6957; // 麟
+		m_table[40667] = 6958; // 黛
+		m_table[40668] = 6959; // 黜
+		m_table[40669] = 6960; // 黝
+		m_table[40672] = 6961; // 黠
+		m_table[40671] = 6962; // 黟
+		m_table[40674] = 6963; // 黢
+		m_table[40681] = 6964; // 黩
+		m_table[40679] = 6965; // 黧
+		m_table[40677] = 6966; // 黥
+		m_table[40682] = 6967; // 黪
+		m_table[40687] = 6968; // 黯
+		m_table[40738] = 6969; // 鼢
+		m_table[40748] = 6970; // 鼬
+		m_table[40751] = 6971; // 鼯
+		m_table[40761] = 6972; // 鼹
+		m_table[40759] = 6973; // 鼷
+		m_table[40765] = 6974; // 鼽
+		m_table[40766] = 6975; // 鼾
+		m_table[40772] = 6976; // 齄
```

Comment:
I guess this is generated from https://github.com/noword/ja2-chinese-resource/blob/master/GB2312_Level1.txt

We should read it from a resource file rather than hard-coding it. I suspect adding 8000 lines could slow down the builds. 

That said, this table just work fine, maybe further improvements can be made in a future PR.

Reply:
Yes, it's very ugly.

Reply:
If we use some resource files, where should them be placed?

Or just provide 5 full arrays in cc file for initializing vector/array. 
It's more efficient when running, not set array's data one by one compared with before.

Reply:
> If we use some resource files, where should them be placed?

I would say put it in a new directory under `assets/externalized`.

Run-time efficiency should not be a big concern. It might take slightly longer (if at all noticeable) to initialize at launch, but we save the compilation time and memory for people not using Chinese mods.

Reply:
Make 5 new files:
```
assets/externalized/EnglishTranslationTable.json
assets/externalized/FrenchTranslationTable.json
assets/externalized/RussianTranslationTable.json
assets/externalized/RussianGoldTranslationTable.json
assets/externalized/SimplifiedChineseTranslationTable.json
```
Then read them dynamically depending on the selected language.
Am I  right?
If we do so, where the `TranslationTable.h` and `TranslationTable.cc` should be placed?
Still at `src/sgp`,  or `src/externalized`?

Reply:
Sounds good to me and I don't think the source particular placement matters much.

Reply:
Done.
Now, the code for reading translation table from json files is in DefaultContentManager.cc.